### ISSUE TITLE
@AirGuardAndroid@Twitter.com to @AirGuardAndroid@mastodon.social

### ISSUE THREAD

**Issue by coastgnu** [seemoo-lab/AirGuard#82]
At least a mirror to mastodon would be good to have.

Regards
CoastGNU

**Comment by Ryu945** [comment]
If they are going to be mirroring then they should consider larger social media networks like Gab as well.

**Comment by luke-jr** [comment]
IMO Gab should just be expected to turn federation back on.

**Comment by Ryu945** [comment]
> IMO Gab should just be expected to turn federation back on.

Gab doesn't have the option to be part of the Mastodon network.   The developers of Mastodon have such an unfounded bias against Gab that they will poison the project code to keep Gab from networking with anyone.   They will also coordinate with large server owners on the Mastodon network and set every Gab server to black list.   No Mastodon user will be able to sync with Gab servers and Gab users will not be able to sync with other servers even if they wanted to.   I am not sure how much US politics your familiar with but people have basically been gaslighted to have severe hatred against certain groups of people they don't know.   Gab is one of those groups of people they have been taught to hate for no reason.    There is 0% chance we can expect the developers of Mastodon network and the server hosters to change their way.   Even if a user on the Mastodon network wanted to talk with someone on Gab, the code writers and server owners will make sure that does not happen.   Mastodon is basically a dead social media platform.   

Gab on the other hand has tried to embrace people of all points of views which is how they have grown so much larger then the Mastodon network.  I would not be surprised if Gab's total active population surpassed Twitter's total active US Only population in a year or two.  I think Gab is roughly 70%-80% US.  That is why I was suggesting Gab as a place because it is a much larger social media network.   There is also far better data privacy on Gab.

There are other issues with Gab syncing with the Mastodon network besides political bias of those in charge of that network.   Gab has grown to have so much more then just a social media platform.    It is an E2EE chat service, a voice chat service ( like Skype) , a video service ( like Youtube) and they have made significant code changes to improve efficiency.  I am not sure how compatible they would be with the Mastodon network.   I suspect it can be made to work but It might be easier for everyone just to make a Gab account.

Something interesting, they made their own payment processor out of necessity because all the US ones have severe political bias so that is another service they have though that is separated from the other Gab services.  I think they are going to partner up with a bank in the future so they will have a bank as well.

I hope this clears up why Gab federating is not an option.   I hope it also shows why federating it worthless.   The small network is not worth federating with and the trouble that small network will give them is not worth it.

**Comment by luke-jr** [comment]
That's false. As much as Mastodon developers don't like it, there are instances that accept federation with Gab. Gab just turned it off to try to force people to use Gab.

Random non-standard features are irrelevant to this use case too.

**Comment by Ryu945** [comment]
> That's false. As much as Mastodon developers don't like it, there are instances that accept federation with Gab. Gab just turned it off to try to force people to use Gab.
> 
> Random non-standard features are irrelevant to this use case too.

Whoever told you that either made it up or got it from someone that made it up.   Did the developers of Mastodon start this to cover up what they did?   I can assure you that Gab's node was viciously forced off the network.  This is actually not the only node that has been forced off the Mastodon network.   They did it with both poisoning the code.  One example of how they poisoned the code is by making apps not able to connect to the Gab node.   They also coordinated with major Mastodon nodes to get them all to add the Gab node to the blacklist.   Federated software should not even have a black list built into it in my opinion but that is another conversation.

This gets back to my point that federation with Mastodon is not an option.   If AirGuardAndroid is going to be making post to more then just Twitter then they should consider other large social media platforms like Gab first and foremost as it is one of the larger alternatives.

**Comment by luke-jr** [comment]
The Mastodon instance I use officially has no interest in blocking Gab, and worked fine with Gab until Gab blocked it (by turning off federation).

There's no reason AirGuard cannot also use an instance that accepts federation from Gab.

**Comment by Ryu945** [comment]
> The Mastodon instance I use officially has no interest in blocking Gab, and worked fine with Gab until Gab blocked it (by turning off federation).
> 
> There's no reason AirGuard cannot also use an instance that accepts federation from Gab.

The problem was that most of the Mastodon network was blocking Gab at the time Gab finally left.

**Comment by luke-jr** [comment]
No, it doesn't. Gab shouldn't complain about them blocking Gab when Gab is blocking literally everyone else.

**Comment by Ryu945** [comment]
> No, it doesn't. Gab shouldn't complain about them blocking Gab when Gab is blocking literally everyone else.

I can only tell you how thing were at the time of Gab leaving the Mastodon network.  It is up to you if you want to believe it.   There are plenty of sites talking about it.   This also doesn't excuse away the developer behavior of poisoning the code.   Anyway, you have my explanation.  Take it if you want to take it.
